projects/components/table/src/subcomponents/table-pagination.component.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild,
+  ViewEncapsulation,
+  AfterContentInit,
+  AfterViewInit,
+} from '@angular/core';
+import { MatPaginator, PageEvent } from '@angular/material/paginator';
+import { MatSelectChange } from '@angular/material/select';
+import { IPsTableIntlTexts } from '@prosoft/components/core';
+import { of, Subject, timer } from 'rxjs';
+import { debounce, takeUntil } from 'rxjs/operators';
+import { PsTableDataSource } from '../data/table-data-source';
+
+@Component({
+  selector: 'ps-table-pagination',
+  template: `
+    <mat-card class="mat-elevation-z0" class="ps-table-pagination__card">
+      <mat-paginator
+        [pageIndex]="pageIndex"
+        [pageSize]="pageSize"
+        [pageSizeOptions]="pageSizeOptions"
+        [length]="dataSource.dataLength"
+        [showFirstLastButtons]="true"
+        (page)="onPage($event)"
+      >
+      </mat-paginator>
+      <div class="mat-paginator-page-size" style="font-size: 12px;">
+        <mat-form-field *ngIf="pages.length > 2" class="mat-paginator-page-size-select">
+          <mat-select id="goToPageSelect" [ngModel]="pageIndex + 1" (selectionChange)="goToPage($event)">
+            <mat-option *ngFor="let page of pages" [value]="page"> {{ page }}</mat-option>
+          </mat-select>
+        </mat-form-field>
+      </div>
+    </mat-card>
+  `,
+  styles: [
+    `
+      .ps-table-pagination__card {
+        display: flex;
+        justify-content: flex-end;
+      }
+    `,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class PsTablePaginationComponent implements OnInit, OnDestroy {
+  public pages: number[] = [];
+  public pageSize: number;
+  public dataLength: number;
+
+  @Input() public dataSource: PsTableDataSource<any>;

```

Comment:
I think it would be better if the datasource wouldn't be used here. Instead you could make the properties you need as inputs. Then you wouldn't need the _internalDetectChanges hack here and the change detection could do its magic.